libs/client-sdk/tests/data_source_status_manager_test.cpp

```diff
@@ -145,6 +147,8 @@ TEST(DataSourceStatusManagerTests, TimeIsUpdatedOnStateChange) {
     status_manager.SetState(DataSourceStatus::DataSourceState::kValid);
 
     auto initial = status_manager.Status().StateSince();
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
TODO(add comment): This test randomly failed because the time hadn't changed at all. Not an ideal way to fix it, but we don't currently have a way to inject the time dependency.